### PR TITLE
Don't allow scheduled rotation without timezone

### PR DESCRIPTION
**Pull request by OneCricketeer**
Got this stacktrace, which is completely unrelated to the code added, but was able to trace the issue via remote debugging, and found the timezone must be added with scheduled rotation. 

```java
java.lang.NullPointerException
	at io.confluent.connect.s3.S3SinkTask.preCommit(S3SinkTask.java:210)
	at org.apache.kafka.connect.runtime.WorkerSinkTask.commitOffsets(WorkerSinkTask.java:345)
	at org.apache.kafka.connect.runtime.WorkerSinkTask.closePartitions(WorkerSinkTask.java:547)
	at org.apache.kafka.connect.runtime.WorkerSinkTask.execute(WorkerSinkTask.java:170)
	at org.apache.kafka.connect.runtime.WorkerTask.doRun(WorkerTask.java:170)
	at org.apache.kafka.connect.runtime.WorkerTask.run(WorkerTask.java:214)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
[2018-02-09 23:01:51,918] (pool-1-thread-1) ERROR - org.apache.kafka.connect.runtime.WorkerTask - WorkerSinkTask{id=abc123-0} Task threw an uncaught and unrecoverable exception
java.lang.NullPointerException
	at io.confluent.connect.s3.S3SinkTask.close(S3SinkTask.java:223)
	at org.apache.kafka.connect.runtime.WorkerSinkTask.commitOffsets(WorkerSinkTask.java:363)
	at org.apache.kafka.connect.runtime.WorkerSinkTask.closePartitions(WorkerSinkTask.java:547)
	at org.apache.kafka.connect.runtime.WorkerSinkTask.execute(WorkerSinkTask.java:170)
	at org.apache.kafka.connect.runtime.WorkerTask.doRun(WorkerTask.java:170)
	at org.apache.kafka.connect.runtime.WorkerTask.run(WorkerTask.java:214)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
```

Also, [HDFS Connect does do this check](https://github.com/confluentinc/kafka-connect-hdfs/blob/master/src/main/java/io/confluent/connect/hdfs/HdfsSinkTask.java#L66).

### CODE REVIEW

**Comment by cricket007** [comment]
@rhauch @wicknicks Can I get a review? 

Doesn't seem one can assign reviewers here.

File: kafka-connect-s3/src/main/java/io/confluent/connect/s3/TopicPartitionWriter.java [hunk 1]
```diff
@@ -127,7 +128,13 @@ public TopicPartitionWriter(TopicPartition tp,
...
         connectorConfig.getLong(S3SinkConnectorConfig.ROTATE_SCHEDULE_INTERVAL_MS_CONFIG);
     if (rotateScheduleIntervalMs > 0) {
-      timeZone = DateTimeZone.forID(connectorConfig.getString(PartitionerConfig.TIMEZONE_CONFIG));
+      String timeZoneString = connectorConfig.getString(PartitionerConfig.TIMEZONE_CONFIG);
+      if (StringUtils.isBlank(timeZoneString)) {
+        throw new ConfigException(PartitionerConfig.TIMEZONE_CONFIG,
+                timeZoneString, "Timezone cannot be empty when using scheduled file rotation."
+        );
+      }
+      timeZone = DateTimeZone.forID(timeZoneString);
```
Comment (wicknicks) [inline kafka-connect-s3/src/main/java/io/confluent/connect/s3/TopicPartitionWriter.java]: @cricket007 I would make this check in S3SinkTask#start() [here](https://github.com/confluentinc/kafka-connect-storage-cloud/blob/master/kafka-connect-s3/src/main/java/io/confluent/connect/s3/S3SinkTask.java#L92). We want to check configuration before the task initiates any real work.

**Comment by cricket007** [comment]
@wicknicks Better?

**Comment by cricket007** [comment]
Breaks on 
https://github.com/confluentinc/kafka-connect-storage-common/commit/5e63b66e43f3793abe423b37dc183d8a05fa66f1 not in master branch yet

**Comment by cricket007** [comment]
@cyrusv Can this be pulled in with the other time-related things

**Comment by kkonstantine** [comment]
@cricket007 I have a PR to fix this in a more general way. Will dig it out from my stash now that we are out of code freeze and update here. Thanks

**Comment by cricket007** [comment]
@kkonstantine Any updates on this one?

**Comment by kkonstantine** [comment]
Closing in favor of https://github.com/confluentinc/kafka-connect-storage-cloud/pull/340 which fixed this during config instantiation, which is more preferable for early validations.

**Pull request closed without merging**